src/RulesetEngine.php

```diff
@@ -4,61 +4,55 @@
 
 namespace SensioLabs\Deptrac;
 
-use SensioLabs\Deptrac\Configuration\ConfigurationRuleset;
-use SensioLabs\Deptrac\Configuration\ConfigurationSkippedViolation;
+use SensioLabs\Deptrac\Configuration\Configuration;
 use SensioLabs\Deptrac\Dependency\Result;
-use SensioLabs\Deptrac\RulesetEngine\RulesetViolation;
+use SensioLabs\Deptrac\RulesetEngine\Context;
 
 class RulesetEngine
 {
-    /**
-     * @return RulesetViolation[]
-     */
-    public function getViolations(Result $dependencyResult, ClassNameLayerResolverInterface $classNameLayerResolver, ConfigurationRuleset $configurationRuleset): array
-    {
-        $violations = [];
+    public function process(
+        Result $dependencyResult,
+        ClassNameLayerResolverInterface $classNameLayerResolver,
+        Configuration $configuration
+    ): Context {
+        $context = new Context();
+
+        $configurationRuleset = $configuration->getRuleset();
+        $configurationSkippedViolation = $configuration->getSkipViolations();
 
         foreach ($dependencyResult->getDependenciesAndInheritDependencies() as $dependency) {
             $layerNames = $classNameLayerResolver->getLayersByClassName($dependency->getClassA());
 
             foreach ($layerNames as $layerName) {
                 $allowedDependencies = $configurationRuleset->getAllowedDependencies($layerName);
 
-                foreach ($classNameLayerResolver->getLayersByClassName($dependency->getClassB()) as $layerNameOfDependency) {
+                $layersNamesClassB = $classNameLayerResolver->getLayersByClassName($dependency->getClassB());
+
+                if (0 === count($layersNamesClassB)) {
```

Comment:
\count or \empty

Reply:
Micro-optimization but feel free to add a php-cs-fixer rule with a PR.